File: Python/007. Reverse Integer.py
class Solution:
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        INT_MAX = 2147483647
        INT_MIN = -2147483648
        tag = False
        if x < 0:
            tag = True
            x = -x
        ans = 0
        while x != 0:
            r = x % 10
            x = x // 10
            ans = ans * 10 + r
        if tag:
            ans = -ans
        if ans > INT_MAX or ans < INT_MIN:
            return 0
        return ans

File: Go/007. Reverse Integer.go
package main

func reverse(x int) int {
	const INT_MAX int64 = 2147483647
	const INT_MIN int64 = -2147483648
	var res int64 = 0
	var symbol int64 = 1
	if x < 0 {
		symbol = -1
		x = -x
	}
	for x != 0 {
		r := x % 10
		res *= 10
		res += int64(r)
		x /= 10
	}
	if symbol * res > INT_MAX || symbol * res < INT_MIN {
		return 0
	}
	return int(symbol * res)
}